src/aind_slims_api/operations/ecephys_session.py

```diff
@@ -0,0 +1,158 @@
+"""
+Module defining operations to build EcephysSession.
+"""
+
+import logging
+from typing import List, Optional
+from pydantic import BaseModel
+from aind_slims_api import SlimsClient
+from aind_slims_api.exceptions import SlimsRecordNotFound
+from aind_slims_api.models.mouse import SlimsMouseContent
+from aind_slims_api.models.ecephys_session import (
+    SlimsMouseSessionResult,
+    SlimsStreamsResult,
+    SlimsStimulusEpochsResult,
+    SlimsExperimentRunStepContent,
+    SlimsExperimentRunStep,
+    SlimsDomeModuleRdrc,
+    SlimsRewardDeliveryRdrc,
+    SlimsRewardSpoutsRdrc,
+    SlimsGroupOfSessionsRunStep,
+    SlimsMouseSessionRunStep,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class EcephysSession(BaseModel):
+    """
+    Pydantic model encapsulating all session-related responses.
+    """
+
+    session_group: Optional[SlimsExperimentRunStep]
```

Comment:
It looks like at least session_group (and maybe session_result?) will always be populated, should they not be Optional?

Reply:
there's a (small) chance that even though there is a mouse session run step in a workflow, that session may not exist in the results table. since we don't have too many examples yet of what is/isn't possible I'll leave session_result as optional. Good catch for session_group though